File: CSharpCodeBase/utils/bhlib/invertor.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class invertor {
 var Inverter = Class({init = function(self, cond)
       this.cond = cond;
       this.returnCode = B_FAILURE;
 }})
 public void Inverter:Behave(creature){
   result = this.cond:Behave(creature) == B_SUCCESS  &&  B_FAILURE  ||  B_SUCCESS;
   this.returnCode = result;
   return this.returnCode;
 }
 return Inverter}}

File: CSharpCodeBase/utils/bhlib/invertor.lua
local Class = require 'utils.hump.class'

local Inverter = Class({init = function(self, cond)
      self.cond = cond
      self.returnCode = B_FAILURE
end})

function Inverter:Behave(creature)
  result = self.cond:Behave(creature) == B_SUCCESS and B_FAILURE or B_SUCCESS
  self.returnCode = result
  return self.returnCode
end

return Inverter